data/dataloader.py

```diff
@@ -172,20 +250,39 @@ def __getitem__(self, index):
 
         """
         # Load image
-        img_link = self.data.iloc[index][SPCData.IMG]
-        img = pil_loader(img_link)
+        img = pil_loader(self.data.iloc[index][SPCData.IMG])
+        ### Padding to Square
+        img_size = img.size
+        img_padding = [0,0]
+        if img_size[0] > img_size[1]:
+            img_padding[1] = int((img_size[0]-img_size[1])/2)
+        else:
+            img_padding[0] = int((img_size[1]-img_size[0])/2)
+        img_padding = tuple(img_padding)
+        img = transforms.Pad(img_padding,fill=0, padding_mode='constant')(img)
+        ###
         img = self.data_transform[self.mode](img)
-        img = rgb_preproc(pil2numpy(img))
-        img = numpy2tensor(img)
-
-        target = self.data.iloc[index][SPCData.LBL]
+#         img = rgb_preproc(pil2numpy(img))
+#         img = pil2numpy(img)
+#         img = numpy2tensor(img)
+        if SPCData.LBL not in self.data.columns:
```

Comment:
im gonna raise a technical debt to remove the requirements of having required columns in the csv files
the main part is in this line i believe and a couple of other spots.